src/frontend/js/components/Search/index.tsx

```diff
@@ -78,49 +74,64 @@ const Search = ({ context }: CommonDataProps) => {
     }
   }, [courseSearchResponse]);
 
+  useEffect(() => {
+    if (filtersPaneClosing) {
+      filtersPaneRef.current?.addEventListener(
+        'transitionend',
+        () => setFiltersPaneClosing(false),
+        { once: true },
+      );
+    }
+  }, [filtersPaneClosing]);
+
   return (
     <div className="search">
       <div
+        role="region"
+        aria-labelledby={`${referenceId}-filters__title`}
+        ref={filtersPaneRef}
         className={`search__filters ${
           !alwaysShowFilters && showFilters ? 'search__filters--active' : ''
         }`}
       >
-        <SearchFiltersPane
-          aria-hidden={alwaysShowFilters ? false : !showFilters}
-          filters={
-            courseSearchResponse?.status === RequestStatus.SUCCESS
-              ? courseSearchResponse.content.filters
-              : null
-          }
-          id={referenceId}
-        />
         {!alwaysShowFilters && (
-          <button
-            aria-expanded={showFilters}
-            aria-controls={referenceId}
-            className="search__filters__toggle"
-            onClick={() => setShowFilters(!showFilters)}
-          >
-            {showFilters ? (
-              <Fragment>
-                <Icon name="icon-cross" className="search__filters__toggle__icon" />{' '}
-                <span className="offscreen">
-                  <FormattedMessage {...messages.hideFiltersPane} />
-                </span>
-              </Fragment>
-            ) : (
-              <Fragment>
-                <Icon name="icon-filter" className="search__filters__toggle__icon" />
-                <span className="offscreen">
-                  <FormattedMessage {...messages.showFiltersPane} />
-                </span>
-              </Fragment>
-            )}
-          </button>
+          <FiltersPaneCloseButton
+            expanded={showFilters}
+            controls={`${referenceId}-filters`}
+            onClick={onFiltersPaneCloseButtonClick}
+          />
+        )}
+        <div
+          className={`search__filters__pane-container ${filtersPaneClosing ? 'is-closing' : ''} ${
+            !alwaysShowFilters && !showFilters && !filtersPaneClosing ? 'is-closed' : ''
+          }`}
+        >
+          <SearchFiltersPane
+            filters={
+              courseSearchResponse?.status === RequestStatus.SUCCESS
+                ? courseSearchResponse.content.filters
+                : null
+            }
+            id={`${referenceId}-filters`}
+          />
+        </div>
+        {/* we repeat the close button at the end because there can be quite a lot of filters,
+				and going back to the close button at the top is cumbersome, especially for screen reader users */}
```

Comment:
There are tabs here instead spaces
```suggestion
        {/* we repeat the close button at the end because there can be quite a lot of filters,
        and going back to the close button at the top is cumbersome, especially for screen reader users */}
```

Reply:
woops!